test/controller_SUITE.erl

```diff
@@ -54,6 +69,47 @@ end_per_testcase(_TestCase, Config) ->
     Config.
 
 %% test cases
+
+each_assignment_is_less_than_or_equal_in_the_input(_) ->
+    Prop = ?FORALL({Total, Counts}, {pos_integer(), sups_with_counts()},
+              begin
+                  SupervisorWithCounts = [ {dummy_pid(), Count} || Count <- Counts ],
+                  {_, Assignments} = amoc_users_sup:distribute(Total, SupervisorWithCounts),
+                  Data = maps:from_list(SupervisorWithCounts),
+                  lists:all(fun({Pid, N}) -> N =< maps:get(Pid, Data) end, Assignments)
+              end),
+    run_prop(?FUNCTION_NAME, Prop, 100, 1).
+
+total_count_is_always_less_than_or_equal_the_requested_amount(_) ->
```

Comment:
we can get rid of this TC if we check `Count =:= min(Total, Available)` in the following TC.